api/client/node/opts.go

```diff
@@ -4,18 +4,37 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/docker/docker/opts"
+	runconfigopts "github.com/docker/docker/runconfig/opts"
 	"github.com/docker/engine-api/types/swarm"
 )
 
 type nodeOptions struct {
+	annotations
 	role         string
 	membership   string
 	availability string
 }
 
+type annotations struct {
+	name   string
```

Comment:
Yes the name is never set. It's not clear how node.annotations.name should be used. So far we are using `hostname` only.